include/umf/proxy_lib_handlers.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_PROXY_LIB_HANDLERS_H
+#define UMF_PROXY_LIB_HANDLERS_H 1
+
+#include <umf/memory_pool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// malloc API handlers
+typedef void (*umf_proxy_lib_handler_free_pre_t)(void *ptr,
+                                                 umf_memory_pool_handle_t pool);
+
+void umfSetProxyLibHandlerFreePre(umf_proxy_lib_handler_free_pre_t handler);
```

Comment:
I think we need to discuss the following question at the UMF technical meeting: Should this API be part of the proxy library or the core UMF library.